databaseutil/databaseutil.go

```diff
@@ -46,6 +49,12 @@ func InsertIntoUsersTable(
 			return nil
 		}
 
+		if pgErr, ok := err.(*pq.Error); ok {
+			if pgErr.Code == "23505" {
+				return UniqueConstraintError
```

Comment:
Considering that in  [create new user](#diff-09499008c3b5f7673ff66923b8758afaR32) we convert this error (regardless of which unique constraint is violated) into a EmailAddressAlreadyInUseError.  Why not throw  EmailAddressAlreadyInUseError here instead.

Reply:
That link doesn't go anywhere for me btw.

`UniqueConstraintError` is a database level error that I expect will be thrown by future functions. I'm expecting that we'll abstract away a lot of postgres errors, because looking up the specific error codes every time is a recipe for disaster

Reply:
fixed the link. I guess it just feels weird throwing an error that we always cast immediately to another error type.